src/app/state/organization/utils/get-main-orgs.ts

```diff
@@ -0,0 +1,40 @@
+export const getMainOrgs = ($getOrgList: any[]) => {
+  const list = [];
+  const subList = [];
+
+  // Separate Parent & Child Orgs
+  $getOrgList.forEach((item: any) => {
+    const namespaceArr = item.namespace.split('.');
+    if (namespaceArr.length === 3) {
+      list.push(item);
+    } else {
+      if (!subList[namespaceArr.length - 4]) {
+        subList[namespaceArr.length - 4] = [];
+      }
+      subList[namespaceArr.length - 4].push(item);
+    }
+  });
+
+  // Remove Unnecessary Sub-Orgs from Main List
+  if (list.length || subList.length) {
+    for (let i = 0; i < subList.length; i++) {
+      const arr = subList[i];
+      if (arr && arr.length) {
+        for (const subOrg of arr) {
+          let exists = false;
+          for (const mainOrg of list) {
+            if (subOrg.namespace && subOrg.namespace.includes(mainOrg.namespace)) {
+              exists = true;
+              break;
+            }
+          }
+          if (!exists) {
+            list.push(subOrg);
+          }
```

Comment:
It seems like you are trying to figure out if a `subOrg`'s parent Org is in the `list`? Is it possible to have different result based on the order of the elements in `arr`? Is you're checking the elements of `list` but also adding to `list` in each iteration?